src/main/java/cloud/orbit/actors/cluster/impl/RedisOrbitClient.java

```diff
@@ -77,31 +78,49 @@ public void subscribe(final String channelId, final MessageListener<Object> mess
 
     private void connectionTask()
     {
-        final boolean nowConnected = redisClient.getNodesGroup().pingAll();
-
-        if(!nowConnected)
+        try
         {
-            isConnected = false;
-        }
+            boolean nowConnected = false;
 
-        if(nowConnected && !isConnected)
-        {
-            boolean subscribedAll = true;
-            for(Pair<String, MessageListener<Object>> subscription : subscriptions)
+            try
             {
-                try
-                {
-                    redisClient.getTopic(subscription.getLeft()).removeAllListeners();
-                    redisClient.getTopic(subscription.getLeft()).addListener(subscription.getRight());
+                nowConnected = redisClient.getNodesGroup().pingAll();
+            }
+            catch (Throwable e)
+            {
+                logger.error("Unexpected error during redis ping", e);
+            }
 
-                } catch(Exception e)
+            if (!nowConnected)
+            {
+                isConnected = false;
+                logger.error("Redis is not connected: {}", redisClient.getNodesGroup().getNodes().stream()
+                        .map((x) -> x.getAddr().toString()).collect(Collectors.joining(",")));
+            }
+
+            if (nowConnected && !isConnected)
+            {
+                boolean subscribedAll = true;
+                for (Pair<String, MessageListener<Object>> subscription : subscriptions)
                 {
-                    subscribedAll = false;
-                    break;
+                    try
+                    {
+                        redisClient.getTopic(subscription.getLeft()).removeAllListeners();
+                        redisClient.getTopic(subscription.getLeft()).addListener(subscription.getRight());
+                    }
+                    catch (Exception e)
+                    {
+                        logger.warn("Could not resubscribe to '{}'\n{}", subscription.getLeft(), e);
```

Comment:
Don't need to `\n` the exception. SLF4J interprets the last parameter properly if it is `Throwable`.

https://www.slf4j.org/faq.html#paramException

Reply:
Really? I never knew that, that's kinda nifty. I will fix.